rustls/src/server/tls13.rs

```diff
@@ -581,40 +581,39 @@ impl CompleteClientHelloHandling {
             .collect();
 
         let supported_groups = suites::KeyExchange::supported_groups();
-        let chosen_group = supported_groups
+
+        let chosen_share = supported_groups
             .iter()
-            .find(|group| share_groups.contains(group))
-            .cloned();
+            .find(|group| share_groups.contains(group)) // Find the chosen group
+            .and_then(|g| shares_ext.iter().find(|k| k.group == *g)); // Then its share
```

Comment:
I'd prefer using `group` over `g` and `share` over `k` -- this code is dense enough as it is...

Reply:
Names changed, good point.